### ISSUE TITLE
Transaction() constructors behave differently

### ISSUE THREAD

**Issue by gagarin55** [ethereum/ethereumj#610]
I've tried different Transaction class constructors and it seems that results are different. Please, check my unit test code. Why its broken ? (Sorrey for Scala, but its almost java)

`

    // 1. create tx and sign it
    val senderPrivKey = "2f9c9d5614ae2c48b0946793b3cc49df0ee9b27a657af7df6b6a782e403e75f1"
    val recipient = "cd2a3d9f938e13cd947ec05abc7fe734df8dd826"

    val nonce = ByteUtil.bigIntegerToBytes(BigInteger.valueOf(0L))
    val gasPrice = ByteUtil.longToBytesNoLeadZeroes(20000000000L)
    val gasLimit = ByteUtil.longToBytesNoLeadZeroes(90000L)
    val receiveAddress = Hex.decode(recipient)
    val value = ByteUtil.bigIntegerToBytes(BigInteger.valueOf(998200000000000000L))
    val chainId = 28.toByte

    val tx = new Transaction(
      nonce,
      gasPrice,
      gasLimit,
      receiveAddress,
      value,
      Array.emptyByteArray,
      chainId)

    val senderPrivateKey = Hex.decode(senderPrivKey)
    tx.sign(ECKey.fromPrivate(senderPrivateKey))

    // 2. save signature
    val signature = tx.getSignature
    val signatureR = BigIntegers.asUnsignedByteArray(signature.r)
    val signatureS = BigIntegers.asUnsignedByteArray(signature.s)
    val signatureV = signature.v

    // 3. create tx with signature

    val tx2 = new Transaction(
      nonce,
      gasPrice,
      gasLimit,
      receiveAddress,
      value,
      Array.emptyByteArray,
      signatureR,
      signatureS,
      signatureV
    )

    assert(Hex.toHexString(tx2.getHash) == Hex.toHexString(tx.getHash))

`

**Comment by zilm13** [comment]
@gagarin55 
val chainId = 28.toByte
if you don't have chainId, use null

**Comment by gagarin55** [comment]
ethereum.getChainIdForNextBlock() gave me 28

**Comment by zilm13** [comment]
@gagarin55 what network do you run it on? (Live, Private, etc)
what version of EthereumJ are you using?

Btw this constructor assumes that you already encoded chainId in V
```java
new Transaction(
  nonce,
  gasPrice,
  gasLimit,
  receiveAddress,
  value,
  Array.emptyByteArray,
  signatureR,
  signatureS,
  signatureV
)
```
We'd better add some doc to it with introducing of chainId

**Comment by Nashatyrev** [comment]
> Btw this constructor assumes that you already encoded chainId in V
@zilm13 Oh, that's not good

**Comment by gagarin55** [comment]
I run on Ropsten Testnet.
I built TX by 1st constructor and than try to restore signed TX by 2nd constructor

**Comment by zilm13** [comment]
@gagarin55 for Ropsten network correct chainId is 3, maybe there are some incorrect Scala type conversions in the middle? I've checked, in Java it's correct.

We will add better Transaction constructor with separate chainId param for the 2nd case very soon.

**Comment by gagarin55** [comment]
Oh, sorry, my failure, Chain ID is 3, sure! But it doesn't matter =) Its still impossible to reconstruct TX by 2nd ctor

**Comment by Nashatyrev** [comment]
@gagarin55 yep, our fault. That was incompatible change. Should be fixed with `1.3.9`: 
https://github.com/ethereum/ethereumj/commit/c1b6de368833a3131a25a9548fdf83ac7359ce84
